### DISCUSSION TITLE
'have.text' and 'have.html' ignore Special characters

### DISCUSSION THREAD

**Discussion by noudadrichem** [cypress-io/cypress#24902] [Questions and Help]
tl;dr; 
Is there any other way to get the 'raw' text/html of a node beside `have.html` ? 

I'm stuck on an assertion that is in a scenario that I have to implement. We're adding hyphenation to text where the HTML will end up looking like this:

![image](https://user-images.githubusercontent.com/16288476/204854497-88fe5ae4-58a9-4143-a64f-e2038d627a62.png)

The problem now is that when using an assertion like this:

```ts
cy.get('summary').should('contain', 'this text will hy­&shy;phen­&shy;ate');
```

Will fail because of this:

![image](https://user-images.githubusercontent.com/16288476/204854757-68ed24aa-7a5b-4c03-a9b4-0111e7e9cbc7.png)

It removes the `&shy;`. So I investigated a bit further. But this also happens with `&amp;` and `&euro;`. 
I'd expect `have.html` to get the 'raw' values of a node. Any flags on this function that I'm missing?

I'm kinda stuck here.. Thanks in advance!
